submit-queue/e2e.go

```diff
@@ -30,9 +30,24 @@ import (
 	github_api "github.com/google/go-github/github"
 )
 
+// PRInfo explicitly collects the fields we wish to encode via JSON.
+type PRInfo struct {
+	Number *int    `json:"number"`
+	URL    *string `json:"html_url"`
```

Comment:
I don't even know what the problems are here, but why not make this an `int` and `string` instead of `*int` and `*string` so you know the PRInfo can't mutate after it's created?